OCPBUGS#6756: Adds conditional updates in web console and some reorg (#63315)
openshift/openshift-docs: modules/before-updating-web-console.adoc

// Module included in the following assemblies:
//
// * updating/updating_a_cluster/updating-cluster-web-console.adoc

:_content-type: CONCEPT
[id="before-updating-web-console{context}"]
= Before updating using the web console

Before updating using the web console, consider the following:

> * Your etcd backup is recent enough in case your update fails and you must restore your cluster to a previous state.

Reviewer: I am not comfortable with suggesting customers restore the cluster from previous etc backup when update status is failing = true. Because failing = true is the indication that update is failing

Reviewer 2: Do you want this list item to be removed?

Author: Let me know what you think. I have updated to "You have recently backed up etcd" we can completely remove though if you would prefer.

> * You might experience an interruption during the update process if you run an Operator or you have configured any application with the pod disruption budget. If `minAvailable` is set to 1 in `PodDisruptionBudget`, the nodes are drained to apply pending machine configs which might block the eviction process. If several nodes are rebooted, all the pods might run on only one node, and the `PodDisruptionBudget` field can prevent the node drain.

Reviewer: Also I do not agree with the line `You might experience an interruption during the update process if you run an Operator or you have configured any application with the pod disruption budget` . I am fine with removing this line altogether.

Reviewer 2: Trevor just created [OSDOCS-7661](https://issues.redhat.com/browse/OSDOCS-7661) to drop this item from this list entirely, would you be willing to wait until I start that ticket to make any changes to this item?

Author: Resolving this so I dont accidentally capture this.

Author: @skopacz1, @LalatenduMohanty and I discussed this. I have removed the sentence requested in this PR, but feel free to edit in however you see fit when you fix the ticket you mentioned.

* You might need to update the cloud provider resources for the new release if your cluster uses manually maintained credentials.

> * You should provide the administrator acknowledgment and review the list of APIs that were removed with the latest Kubernetes release, and migrate any affected components to use the new API version.

Reviewer 3: The outgoing:

> * Review the list of APIs that were removed in Kubernetes 1.27, migrate any affected components to use the new API version, and provide the administrator acknowledgment. For more information, see xref:../../updating/preparing_for_updates/updating-cluster-prepare.adoc#updating-cluster-prepare[Preparing to update to {product-title} 4.14].

was possibly more version-specific than we'd like, but I prefer it's opening that focuses on reviewing ack requests instead of providing ack approvals.  We want to reduce the chances that folks blindly click through the acks without actually performing the assessments that the acks are requesting.  Maybe:

* You should review administrator acknowledgement requests, take any recommended actions, and provide the acknowledgement when you are ready.

or something, to keep the opening focus on figuring out what you're being asked to ack?

Admin acks are also a subset of `Upgradeable=False` conditions, so I'm personally fine not discussing admin acks at all here, and instead asking the admin to address any `Upgradeable=False` conditions that the UI might be presenting.

Author: Updated! Thanks for the suggestion.

> *  You can perform a partial update by updating the worker or custom pool nodes to accommodate the time it takes to update. You can pause and resume within the progress bar of each pool.

Reviewer 3: nit: seems like you have an extra space here in `*  You`

// Module included in the following assemblies:
//
// * updating/updating_a_cluster/updating-cluster-web-console.adoc

:_content-type: CONCEPT
[id="before-updating-web-console{context}"]
= Before updating using the web console

Before updating using the web console, you should:

> * Have access to the cluster as a user with `cluster-admin` privileges.

Reviewer 2: This can probably be removed since it's already a prereq in all the modules that need it (though I'd double check whether it's also a necessary prereq for the module about changing the update server)

> * Have a recent etcd backup in case your update fails and you must restore your cluster to a previous state.

Reviewer 2: I would double check with an SME, but I think this prereq might be moot if there's nothing the user could do to trigger a new backup. If that's true, this is more like "hope that your most recent backup is recent enough to revert to if disaster recovery is necessary."

One idea might be to keep this here but reword it to something like:
"Check to see that your etcd backup is recent enough in case your update fails and you must restore your cluster to a previous state." Plus maybe a sentence about how etcd periodically creates a new backup on its own?

Author: Updated to your suggestion, but wondering if we should specify what "recent enough" means. Leaving this comment thread open to discuss with SMEs.

> * You must replace {op-system-base}7 workers with {op-system-base}8 or {op-system} workers before updating to {product-title} {product-version}. Red Hat does not support in-place {op-system-base}7 to {op-system-base}8 updates for {op-system-base} workers; those hosts must be replaced with a clean operating system install.

Reviewer 2: I'd move this to prereqs for update modules (and ensure parallel wording with other prereqs)

> * Ensure all Operators previously installed through Operator Lifecycle Manager (OLM) are updated to their latest version in their latest channel. Updating the Operators ensures they have a valid update path when the default OperatorHub catalogs switch from the current minor version to the next during a cluster update.

Reviewer 2: I'd move this to prereqs for update modules (and ensure parallel wording with other prereqs)

> * Ensure that all machine config pools (MCPs) are running and not paused. Nodes associated with a paused MCP are skipped during the update process. You can pause the MCPs if you are performing a canary rollout update strategy.

Reviewer 2: I'd move this to prereqs for update modules and reword as follows:

"Ensure that all machine config pools (MCPs) are running and not paused, unless you are performing a canary rollout update strategy. Nodes associated with a paused MCP are skipped during the update process."

//remove this???^ or maybe just add another bullet that you can break up the update?

> * To accommodate the time it takes to update, you are able to do a partial update by updating the worker or custom pool nodes. You can pause and resume within the progress bar of each pool.

Reviewer 2: ```suggestion
* To accommodate the time it takes to update, you can perform a partial update by updating the worker or custom pool nodes. You can pause and resume within the progress bar of each pool.
```

> * If your cluster uses manually maintained credentials, update the cloud provider resources for the new release.

Reviewer 2: I'd move this to prereqs for update modules (and ensure parallel wording with other prereqs)

> * Review the list of APIs that were removed in Kubernetes 1.26, migrate any affected components to use the new API version, and provide the administrator acknowledgment.

Reviewer 2: I'd move this to prereqs for update modules (and ensure parallel wording with other prereqs)

I think this one is tricky because it might not be in each version's prereqs, and the K8s version is probably different in each OCP version this appears in.

Author: I think there is always a list of APIs that are removed each release?? I think we could generalize it and keep it under the considerations module as "check the APIs that were removed" maybe

//may need to move to a troubleshooting section

> //* If you run an Operator or you have configured any application with the pod disruption budget, you might experience an interruption during the update process. If `minAvailable` is set to 1 in `PodDisruptionBudget`, the nodes are drained to apply pending machine configs which might block the eviction process. If several nodes are rebooted, all the pods might run on only one node, and the `PodDisruptionBudget` field can prevent the node drain.

Reviewer 2: If you do end up moving those other prereqs into relevant modules and this module only has a list of "considerations to make", this would be a good item to uncomment and include in the list above
